float64 distance
float64 y_rotation
float64 z_rotation
